.github/workflows/build_and_test.yml

```diff
@@ -15,32 +15,47 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  check_code_changes:
+    name: Check Code Changes
+    uses: ./.github/workflows/_check_code_changes.yml
+    with:
+      event_name: ${{ github.event_name }}
+      # For pull_request, use PR's base and head. For push, use event's before and sha.
+      base_sha: ${{ github.event_name == 'pull_request' && github.event.pull_request.base.sha || github.event.before }}
```

Comment:
I think `github.event.pull_request.base.sha` isn't quite right. On the "Check Code Changes" step of this PR, I can see the following output:

![screenshot-2025-05-09-19-19-36](https://github.com/user-attachments/assets/ed80db24-da1b-4f9d-b888-b9ff52527015)

IOW, GitHub thinks that this PR changes some `.cpp` file, even if that is not the case.

I think that's because `base.sha` just means the SHA of the main branch at the time of running the check. Instead, we probably want the SHA of the commit right before the first commit of the PR in the author's source branch.

Reply:
Updated!